File: api/cron.py
#!/usr/bin/env python3
import os
import sys
from pathlib import Path

sys.path.append(str(Path(__file__).resolve().parent.parent))

import get_cartoon
import send_cartoon

def handler(req, res):
    cron_secret = os.getenv('CRON_SECRET')
    authorization_header = req.headers.get('Authorization')

    if authorization_header != f"Bearer {cron_secret}":
        return res.status(401).send('Unauthorized')
    
    cartoon_url, cartoon_date = get_cartoon.get_latest_cartoon()
    print(cartoon_url, cartoon_date)
    
    if cartoon_date and get_cartoon.is_new_cartoon(cartoon_date):
        send_cartoon.send_message(cartoon_url)
    
    res.status(200).send("Cron job executed successfully.")

if __name__ == "__main__":
    handler()

File: api/cron.js
const { exec } = require('child_process');

export default function handler(req, res) {
  const authHeader = req.headers.get('authorization');
  if (
    !process.env.CRON_SECRET ||
    authHeader !== `Bearer ${process.env.CRON_SECRET}`
  ) {
    return res.status(401).json({ success: false, message: 'Unauthorized' });
  }

  exec('python3 api/cron.py', (error, stdout, stderr) => {
    if (error) {
      console.error(`Error executing script: ${error.message}`);
      return res.status(500).json({ success: false, message: 'Internal Server Error' });
    }
    if (stderr) {
      console.error(`Script stderr: ${stderr}`);
      return res.status(500).json({ success: false, message: 'Internal Server Error' });
    }
    console.log(`Script stdout: ${stdout}`);
    res.status(200).json({ success: true, message: 'Cron job executed successfully' });
  });
}
